src/openrct2/ride/Vehicle.cpp

```diff
@@ -9478,38 +9478,38 @@ static void vehicle_update_track_motion_powered_ride_acceleration(
     };
 
     uint8_t speedModifier = FULL_SPEED;
-    uint16_t trackType = vehicle->track_direction >> 2;
+    uint16_t trackType = track_type >> 2;
 
     if (trackType == TRACK_ELEM_LEFT_QUARTER_TURN_1_TILE)
     {
-        speedModifier = (vehicle->TrackSubposition == VEHICLE_TRACK_SUBPOSITION_GO_KARTS_LEFT_LANE) ? HALF_SPEED
+        speedModifier = (TrackSubposition == VEHICLE_TRACK_SUBPOSITION_GO_KARTS_LEFT_LANE) ? HALF_SPEED
                                                                                                     : THREE_QUARTER_SPEED;
     }
     else if (trackType == TRACK_ELEM_RIGHT_QUARTER_TURN_1_TILE)
     {
-        speedModifier = (vehicle->TrackSubposition == VEHICLE_TRACK_SUBPOSITION_GO_KARTS_RIGHT_LANE) ? HALF_SPEED
+        speedModifier = (TrackSubposition == VEHICLE_TRACK_SUBPOSITION_GO_KARTS_RIGHT_LANE) ? HALF_SPEED
                                                                                                      : THREE_QUARTER_SPEED;
     }
 
-    uint8_t speed = vehicle->speed;
+    uint8_t curSpeed = speed;
```

Comment:
technically this is the `newSpeed`

Reply:
Not really. There is no assignment from curSpeed to speed.

The value is calculated to determine how big the acceleration will be. It is based on the current speed value and on the track.

I've renamed it to 'modifiedSpeed' and extracted a function for clarity.